motioneye/mjpgclient.py

```diff
@@ -278,53 +288,50 @@ def _seek_content_length(self):
         r_future.add_done_callback(self._on_before_content_length)
 
     def _on_before_content_length(self, future: Future):
-        try:
-            future.result()
-        except Exception as e:
-            self._error(e)
-        else:
-            if self._check_error():
-                return
+        result, _ = self._get_future_result(future)
+        if not result:
+            return
+
+        if self._check_error():
+            return
 
-            r_future = utils.cast_future(self.read_until(b'\r\n\r\n'))
-            r_future.add_done_callback(self._on_content_length)
+        r_future = utils.cast_future(self.read_until(b'\r\n\r\n'))
+        r_future.add_done_callback(self._on_content_length)
 
     def _on_content_length(self, future: Future):
-        try:
-            data = future.result()
-        except Exception as e:
-            self._error(e)
-        else:
-            if self._check_error():
-                return
-
-            matches = re.findall(rb'(\d+)', data)
-            if not matches:
-                self._error(
-                    'could not find content length in mjpg header line "{header}"'.format(
-                        header=data
-                    )
+        result, data = self._get_future_result(future)
+        if not result:
+            return
+
+        if self._check_error():
+            return
```

Comment:
```suggestion
        if not result or self._check_error():
            return
```
If hope it is evaluated as
```py
if (not result) or self._check_error()
```
instead of
```py
if not (result or self._check_error())
```
😅